test/api-documentation.test.js

```diff
@@ -571,4 +589,237 @@ describe('<api-documentation>', function() {
+            });
+
+            it('should hide api-server-selector', () => {
+              assert.isTrue(serverSelector.hidden);
+            });
+          });
+
+          describe('when selecting a slot server', () => {
+            beforeEach(() => {
+              const event = {
+                detail: {
+                  selectedValue: 'http://customServer.com',
+                  selectedType: 'slot'
+                }
+              };
+              element.dispatchEvent(new CustomEvent('api-server-changed', event));
+            });
+
+            it('should update selectedServerValue and selectedServerType', () => {
+              assert.equal(element.selectedServerValue, 'http://customServer.com');
+              assert.equal(element.selectedServerType, 'slot');
+            });
+          });
+
+          describe('when selecting a custom base uri', () => {
+            beforeEach(async () => {
+              const event = {
+                detail: {
+                  selectedValue: 'https://www.google.com',
+                  selectedType: 'custom'
+                }
+              };
+              element.dispatchEvent(new CustomEvent('api-server-changed', event));
+            });
+
+            it('should update selectedServerValue and selectedServerType', () => {
+              assert.equal(element.selectedServerValue, 'https://www.google.com');
+              assert.equal(element.selectedServerType, 'custom');
+            });
+
+            describe('and then clearing the selection', () => {
+              beforeEach(() => {
+                const event = {
+                  detail: {
+                    selectedValue: undefined,
+                    selectedType: undefined
+                  }
+                };
+                element.dispatchEvent(new CustomEvent('api-server-changed', event));
+              });
+
+              it('should update selectedServerValue and selectedServerType', () => {
+                assert.equal(element.selectedServerValue, undefined);
+                assert.equal(element.selectedServerType, undefined);
+              });
+
+              describe('and then selecting an existing server', () => {
+                beforeEach(() => {
+                  const event = {
+                    detail: {
+                      selectedValue: 'http://{instance}.domain.com/',
+                      selectedType: 'server'
+                    }
+                  };
+                  element.dispatchEvent(new CustomEvent('api-server-changed', event));
+                });
+
+                it('should update selectedServerValue and selectedServerType', () => {
+                  assert.equal(element.selectedServerValue, 'http://{instance}.domain.com/');
+                  assert.equal(element.selectedServerType, 'server');
+                });
+              });
+            });
+          });
+
+          describe('when navigating to a method', () => {
```

Comment:
Please, remove "when" from the name :)
I will write a doc about naming convention, someday.